cmd/tunneld/tunneld.go

```diff
@@ -144,14 +146,17 @@ func fatal(format string, a ...interface{}) {
 }
 
 func startAutocert(opts *options, server *tunnel.Server, logger log.Logger) {
-	cacheDir := filepath.Join(os.Getenv("HOME"), ".cache", "autocerts")
+	cacheDir := cacheDir(opts.letsEncryptCacheDir)
 	m := &autocert.Manager{
 		Prompt:     autocert.AcceptTOS,
-		HostPolicy: autocert.HostWhitelist(strings.Split(opts.secureHosts, ",")...),
+		HostPolicy: hostPolicy(server),
 		Cache:      autocert.DirCache(cacheDir),
 	}
+	// Allow to bind to a specific host ignoring ports.
+	httpAddr := fmt.Sprintf("%s:80", trimPort(opts.httpsAddr))
```

Comment:
What's the point in running HTTP with Let's Encrypt?

Reply:
I think that it should be validated and if user provides the `letsEncrypt` flag and `httpAddr` or `httpsAddr` with a custom port an error shall be returned and process terminated.

Reply:
Validation will be added, as for running the HTTP it is required by lets encrypt, when validating the domain Let's Encrypt makes a request to `http://domain/.well-known/acme-challenge/{challenge-token}` and the server must respond to the cllange, this is handled by the manager with a fallback handler, we can also enforce https by redirecting to https if LE is enabled.

Reply:
IMO if the flag it true then the HTTP handler should just redirect to HTTPS.

Reply:
That is except for the challenge.